src/library/export/coverartcopyworker.cpp

```diff
@@ -0,0 +1,102 @@
+#include "library/export/coverartcopyworker.h"
+
+#include <QBuffer>
+#include <QDebug>
+#include <QFileInfo>
+#include <QMessageBox>
+
+void CoverArtCopyWorker::run() {
+    copyFile(m_coverArtImage, m_coverArtCopyPath);
+
+    if (m_bStop.loadAcquire()) {
+        emit canceled();
+        return;
+    }
+}
+
+void CoverArtCopyWorker::copyFile(
+        const QImage& m_coverArtImage,
+        const QString& m_coverArtCopyPath) {
+    QFileInfo dest_fileinfo(m_coverArtCopyPath);
+
+    if (dest_fileinfo.exists()) {
+        switch (makeOverwriteRequest(m_coverArtCopyPath)) {
```

Comment:
Does this call a dialog from a worker thread? 
I am not sure if this works under all circumstances.
At least it is more clean to do GUI stuff from the main thread.

Reply:
Ah I see this is the pattern used in the export worker. So it is probably OK.